### ISSUE TITLE
`aria-label="Dialog"` inside the Dialog is not accessible enough

### ISSUE THREAD

**Issue by GomezIvann** [dxc-technology/halstack-react#2139]
We are falling short in the accessibility of the Dialog component, we need to review this and find a solution following the W3C patterns which recommend the usage of `aria-labelledby` and a title inside the Dialog: https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/.
